patmat/src/main/scala/patmat/Huffman.scala

```diff
@@ -105,15 +130,24 @@ trait Huffman extends HuffmanInterface:
    * In such an invocation, `until` should call the two functions until the list of
    * code trees contains only one single tree, and then return that singleton list.
    */
-  def until(done: List[CodeTree] => Boolean, merge: List[CodeTree] => List[CodeTree])(trees: List[CodeTree]): List[CodeTree] = ???
+  def until(done: List[CodeTree] => Boolean, merge: List[CodeTree] => List[CodeTree])(trees: List[CodeTree]): List[CodeTree] = {
+    if(done(trees)){
+      trees
+    }else{
+      until(done, merge)(merge(trees))
+    }
```

Comment:
We usually space those
```suggestion
    if (done(trees)) {
      trees
    } else {
      until(done, merge)(merge(trees))
    }
```